bolt-modules/dir/lib/puppet/functions/dir/children.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+# Returns an array containing all of the filenames except for "." and ".." in the given directory.
+Puppet::Functions.create_function(:'dir::children', Puppet::Functions::InternalFunction) do
+  # @param dirname Absolute path or Puppet module name.
+  # @return Array of files in the given directory.
+  # @example List filenames from an absolute path.
+  #   dir::children('/home/user/subdir/')
+  # @example List filenames from a Puppet file path.
+  #   dir::children('puppet_agent')
+  dispatch :children do
+    scope_param
+    required_param 'String', :dirname
+    return_type 'Array'
+  end
+
+  def children(scope, dirname)
+    # Send Analytics Report
+    Puppet.lookup(:bolt_executor) {}&.report_function_call(self.class.name)
+    puppet_mod = scope.compiler.environment.module(dirname)&.path
```

Comment:
+1 for being able to specify a module by name. I think this could allow a little more flexibility by either allowing you to specify a path within a module (e.g. `module/plans`, `module/files`) or limit it to loading children in the module's `files` directory.